src/gsy_e/models/strategy/market_agents/one_sided_engine.py

```diff
@@ -165,11 +164,6 @@ def event_offer_traded(self, *, trade):
 
         if trade.offer_bid.id == offer_info.target_offer.id:
             # Offer was accepted in target market - buy in source
-            source_rate = offer_info.source_offer.energy_rate
-            target_rate = offer_info.target_offer.energy_rate
-            assert abs(source_rate) <= abs(target_rate) + FLOATING_POINT_TOLERANCE, \
```

Comment:
Why remove this assertion? I find it very valuable in order to prevent bugs in the chain trade operation.

Reply:
@Hassan754 I am aware that this condition can fail for the case of degrees of freedom, but I still think that these asserts can remain for the default case. You can have an if check before the assert in order to circumvent it if any dof has been set. That way the check will still apply for non-dof trades.